### DISCUSSION TITLE
decoder in espnow receiver

### DISCUSSION THREAD

**Discussion by aqildad-create** [pschatzmann/arduino-audio-tools#767] [Q&A] [answered]
i send acknowledgment like below its working and playing but their are sound jerks which is because delay after decoder write and new data coming from sender (may be ) . how to overcome this ?

this is receiver handler were i write to decoder and then send acknowledgement to sender. then sender send new data . 

Is this correct way ? if yes then  am getting small sound breaks . how to overcome this ?  

OnDataRecv(
.
.
decStreammp3.write((uint8_t *)Filedatas.charF , Filedatas.FiLEDATAsze);
esp_now_send(broadcastAddres, (uint8_t *) &outgoingSetpoints, sizeof(outgoingSetpoints)); //acnowledgment send

}

**Comment by pschatzmann** [comment, marked as answer]
I don't understand why you do this: acknowledgements are part of the protocol itself!

**Reply by aqildad-create** [reply]
as per my understanding their is send acknowledgment of espnow .  but if i send new data with sender acknowledgment then it is very is very fast as compare to decoder write stream .

**Reply by pschatzmann** [reply]
That's hard to belief: the confirmation can only be only sent when the decoder write stream has been processed: It will block if you have I2S as final output in your chain.

If you try to implement your own logic I would try to keep a buffer filled on the receiving side. The buffer needs to be big enough to bridge any delays introduced by your processing.
